static/js/lib/api.js

```diff
@@ -92,12 +93,17 @@ import { fetchWithCSRF } from './api';
  *  - non 2xx status codes will reject the promise returned
```

Comment:
And also add a comment explaining about the conversion from maybe to promise in case someone isn't familiar with sanctuary